### PR TITLE
[MM-47896] E2E tests for GM and Channel name collision

### PR DESCRIPTION
**Pull request by mylonsuren**
#### Summary

Fixes e2e tests and adds an extra test for GM name collision.

Fixed existing tests:
- MM-T884_2

Added a new test:
- MM-T884_3

#### Ticket Link

[MM-47896](https://mattermost.atlassian.net/browse/MM-47896)

#### Related Pull Requests
- Has a server PR https://github.com/mattermost/mattermost-server/pull/21639

#### Release Note

```release-note
NONE
```

[MM-47896]: https://mattermost.atlassian.net/browse/MM-47896?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by mylonsuren** [comment]
@jgilliam17 This addresses the E2E failures in https://github.com/mattermost/mattermost-server/pull/21639, and also adds another test to cover the new case.

**Comment by jgilliam17** [comment]
Thanks @mylonsuren 
New tests look good and passing on the [E2E report](https://automation-dashboard.vercel.app/cycles/6bfa6e86-1be6-45ba-ae83-267c35105f56)
I added a PR to update this test case in Zephyr to reflect new changes.
Screen Shot 2022-12-19 at 11 53 10 AM

**Review by jgilliam17 — APPROVED** [review]

**Comment by jgilliam17** [comment]
Added `do not merge` label until related PR is merged.

**Comment by mylonsuren** [comment]
@saturninoabril Gentle ping to review this when you get a chance 🙂

**Comment by saturninoabril** [comment]
Hello Mylon, sorry for the delay. I'll take a look.

File: e2e/cypress/tests/integration/channel_settings/channel_name_validations_spec.ts
Comment (saturninoabril) [inline e2e/cypress/tests/integration/channel_settings/channel_name_validations_spec.ts:67]: This is failing on me. Is it correct the text was changed as above?

File: e2e/cypress/tests/integration/channel_settings/channel_name_validations_spec.ts
Comment (saturninoabril) [inline e2e/cypress/tests/integration/channel_settings/channel_name_validations_spec.ts:86]: This test is failing on me. It's successfully creating the channel.

**Review by saturninoabril — CHANGES_REQUESTED** [review]
Hello @mylonsuren, could you check once again if all the tests in `channel_name_validations_spec.ts` passed on you.

File: e2e/cypress/tests/integration/channel_settings/channel_name_validations_spec.ts
Comment (mylonsuren) [reply e2e/cypress/tests/integration/channel_settings/channel_name_validations_spec.ts:86]: @saturninoabril For the two failing tests you're seeing, was it using https://github.com/mattermost/mattermost-server/pull/21639 as the server branch?

Comment (saturninoabril) [reply e2e/cypress/tests/integration/channel_settings/channel_name_validations_spec.ts:86]: Right, that's it. All passed now. Thanks!

**Review by saturninoabril — CHANGES_REQUESTED** [review]
Test flow looks good to me, except for properly using queries for this PR and for future reference.

Whenever possible, please use the queries being used for Accessibility. See our docs on what queries to use in order - https://developers.mattermost.com/contribute/more-info/webapp/e2e/which-query-to-use/. We're not fully compliant there yet but that's what we're aiming for.

The idea behind is to test the way our application is used, where in most cases users (human or accessibility devices) interact with text and not via element ID or CSS locators. Doing so will give us more confidence in our testing.

**Comment by mylonsuren** [comment]
> Test flow looks good to me, except for properly using queries for this PR and for future reference.
> 
> Whenever possible, please use the queries being used for Accessibility. See our docs on what queries to use in order - https://developers.mattermost.com/contribute/more-info/webapp/e2e/which-query-to-use/. We're not fully compliant there yet but that's what we're aiming for.
> 
> The idea behind is to test the way our application is used, where in most cases users (human or accessibility devices) interact with text and not via element ID or CSS locators. Doing so will give us more confidence in our testing.

Thanks for the feedback @saturninoabril, I'll keep this in mind in when writing future E2Es! I've made the changes you requested, so should be good for another review 😄

**Review by saturninoabril — APPROVED** [review]
Thanks Mylon! LGTM.

**Comment by saturninoabril** [comment]
The daily test is failing after this was merged since the corresponding [server PR](https://github.com/mattermost/mattermost-server/pull/21639) is still pending review.

**Pull request merged**
